pkg/scheduler/nodeinfo/node_info.go

```diff
@@ -471,11 +482,23 @@ func (n *NodeInfo) Clone() *NodeInfo {
 // VolumeLimits returns volume limits associated with the node
 func (n *NodeInfo) VolumeLimits() map[v1.ResourceName]int64 {
```

Comment:
As a relatively low-hanging optimization fruit, volume limits for each node can be pre-calculated in informer callbacks (`AddCSINode` / `UpdateCSINode` + their `Node` equivalents). 
Profiler says that `VolumeLimits()` call itself takes 3s out of 8s of whole `attachableLimitPredicate()`.

Furthermore, list of migrated plugins can be pre-parsed for each node there.

I will defer to sig-scheduling to judge if current performance is OK or it needs to be fixed as part of this PR.

Reply:
I'm surprised. The number of objects being iterated over per call should not be very large.  I would have suspected the pod iteration to take the most cycles.

Reply:
@jsafrane Is the perf overhead applied in regular case (without PV/PVC involved)?

Reply:
> I'm surprised. The number of objects being iterated over per call should not be very large

It is not large, but it is called for every combination of Node and Pod, which is lot of calls. Even simple feature gate check there takes ~0.5s out of this 8 seconds increase.

> Is the perf overhead applied in regular case (without PV/PVC involved)?

Good question! Yes, **all pods with volumes (incl. Secrets) gets such overhead, if CSINode object exists**.

In theory, any volume can be migrated to CSI, so we must know limits reported by the driver. In practice, we won't migrate ephemeral volumes like Secrets, ConfigMap, EmptyDir and so on, some (dirty?) shortcuts are possible. For example, caching of `VolumeLimits` value in `NodeInfo` as I suggested above.

Reply:
>  Profiler says that VolumeLimits() call itself takes 3s out of 8s of whole attachableLimitPredicate().

To be clear, 8s is avg time taken by `VolumeLimits()` per scheduling of a pod?